rascal-vscode-extension/src/fs/RascalFileSystemProviders.ts

```diff
@@ -47,6 +47,27 @@ export class RascalFileSystemProvider implements vscode.FileSystemProvider {
         });
     }
 
+    sendRequest<R>(uri : vscode.Uri, method: string): Promise<R>;
+    sendRequest<R, A>(uri : vscode.Uri, method: string, param: A): Promise<R>;
+    sendRequest<R, A>(uri : vscode.Uri, method: string, param?: A): Promise<R> {
+        return this.client.sendRequest<R>(method, param ?? { uri: uri.toString()} )
+            .catch((r: ResponseError) => {
+                if (r !== undefined) {
+                    this.client.info("Got responds error from the file system: ", r);
+                    switch (r.code) {
+                        case -1: throw vscode.FileSystemError.FileExists(uri);
```

Comment:
There are no constants for these that we can use?

Reply:
sadly no. I mean I would have to repeat the constants in both java and typescript, so their added benefit would be very low.